himbaechel/uarch/gowin/pack.cc

```diff
@@ -2981,6 +2981,37 @@ struct GowinPacker
         }
     }
 
+    // ===================================
+    // HCLK -- CLKDIV and CLKDIV2 for now
+    // ===================================
+    void pack_hclk(void)
+    {
+        log_info("Pack HCLK cells...\n");
+
+        for (auto &cell : ctx->cells) {
+            auto ci = cell.second.get();
+            if (ci->type == id_CLKDIV) {
+                NetInfo *in = ci->getPort(id_HCLKIN);
+                CellInfo *this_driver = in->driver.cell;
```

Comment:
this should probably have some error handling for the case of `HCLKIN` being disconnected/undriven, i.e. either `in` or `in->driver.cell` being `nullptr`